tests/components/recorder/test_migration_from_schema_32.py

```diff
@@ -39,6 +40,17 @@
 ORIG_TZ = dt_util.DEFAULT_TIME_ZONE
 
 
+async def _wait_migration_done(hass: HomeAssistant) -> None:
+    """Wait for the migration to be done."""
+    migration_tasks = (
+        4  # 2 for the task, and 2 in case there are additional rows to migrate
+    )
+    for _ in range(migration_tasks):
+        await asyncio.sleep(0.35)
```

Comment:
This is not a good solution. I only added it to verify this is a race and not an actual behavior change in sqlalchemy

Reply:
If the CI passes with this change, I'll work on an alternate solution. Sadly the test is passing locally so this will be hard to figure out a good fix without a few long iterations